libraries/AP_RangeFinder/AP_RangeFinder.cpp

```diff
@@ -796,9 +796,22 @@ void RangeFinder::Log_RFND() const
 bool RangeFinder::prearm_healthy(char *failure_msg, const uint8_t failure_msg_len) const
 {
     for (uint8_t i = 0; i < RANGEFINDER_MAX_INSTANCES; i++) {
-        if (((Type)params[i].type.get() != Type::NONE) && (drivers[i] == nullptr)) {
-          hal.util->snprintf(failure_msg, failure_msg_len, "Rangefinder %X was not detected", i + 1);
-          return false;
+        if ((Type)params[i].type.get() != Type::NONE) {
+            if(drivers[i] == nullptr) {
+                hal.util->snprintf(failure_msg, failure_msg_len, "Rangefinder %X was not detected", i + 1);
+                return false;
+            }
+
+            switch (drivers[i]->status()) {
+            case Status::NoData:
+                hal.util->snprintf(failure_msg, failure_msg_len, "No data received from Rangefinder %X", i + 1);
+                return false;
+            case Status::NotConnected:
+                hal.util->snprintf(failure_msg, failure_msg_len, "Rangefinder %X is not connected", i + 1);
+                return false;
+            default:
+                break;
```

Comment:
```suggestion
            case Status::OutOfRangeLow:
            case Status::OutOfRangeHigh:
            case Status::Good:
                break;
```

If we add a status this makes people think about whether it needs to be added here.